app/services/ui_notifications/clean_expired.rb

```diff
@@ -0,0 +1,40 @@
+module UINotifications
+  class CleanExpired
+    def initialize(options = {})
+      if options[:blueprint].present?
+        blueprint = NotificationBlueprint.find_by(name: options[:blueprint])
+        raise(Foreman::Exception, 'blueprint name does not exist') if blueprint.nil?
+        @blueprint = blueprint.name
+      end
+      parsed_time = options[:expired_at].is_a?(String) ? options[:expired_at].to_time : Time.now.utc
+      raise(Foreman::Exception, 'Expired time cannot be greater then now') if parsed_time > Time.now.utc
+      @expired_at = parsed_time
+      @group = options[:group]
+    end
+
+    def clean!
+      UINotifications::Base.logger.info("Removing all expired notifications")
+      to_delete = Notification.joins(:notification_blueprint)
+                      .where('notifications.expired_at < ?', @expired_at)
+                      .where(blueprint_query)
+                      .where(group_query)
+      @deleted_count = to_delete.destroy_all.length
```

Comment:
`destroy_all` instantiates all objects and runs callbacks when destroying. While this is fine for a small amount, this can be quite slow if there is a very large number of records to destroy - in that case it would be faster to use delete_all to directly delete everything from the database (like we do for report expiry for example).